Shuttle-front/src/app/estimation-map/estimation-map.component.ts

```diff
@@ -0,0 +1,182 @@
+import { MapEstimationService } from '../services/map/map-estimation.service';
+
+const iconRetinaUrl = 'assets/marker-icon-2x.png';
+const iconUrl = 'assets/marker-icon.png';
+const shadowUrl = 'assets/marker-shadow.png';
+
+const carMarkerUrl = 'assets/car-marker.png';
+const iconDefault = L.icon({
+  iconRetinaUrl,
+  iconUrl,
+  shadowUrl,
+  iconSize: [25, 41],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  tooltipAnchor: [16, -28],
+  shadowSize: [41, 41]
+});
+
+const carMarkerIcon = L.icon({
+  iconUrl: carMarkerUrl,
+  shadowUrl,
+  iconSize: [48, 48],
+  iconAnchor: [12, 41],
+  popupAnchor: [1, -34],
+  tooltipAnchor: [16, -28],
+  shadowSize: [41, 41]
+});
+L.Marker.prototype.options.icon = iconDefault;
+
+@Component({
+  selector: 'app-estimation-map',
+  templateUrl: './estimation-map.component.html',
+  styleUrls: ['./estimation-map.component.css']
+})
+export class EstimationMapComponent implements AfterViewInit, OnChanges {
+
+  @Input()
+  currentRoute: [String, String] = ["", ""];
+  
+
+  constructor(private mapService: MapEstimationService){
+    this.pull = interval(3 * 1000).pipe(startWith(0)).subscribe(() => {
+      this.refreshActiveDrivers();
+    });
+    this.driverLocationMarkers = [];
+  }
+
+
+  private map?: L.Map;
+  private destinationMarker?: L.Marker;
+  private departureMarker?: L.Marker;
+  private routeControl?: L.Routing.Control;
+
+  private destinationCoordinates?: L.LatLng;
+  private departureCoordinates?: L.LatLng;
+
+  private pull: Subscription;
+  private driverLocationMarkers: Array<L.Marker>;
+
+  private initMap(): void{
+    this.map = L.map("estimation-map", {
+      center: [ 45.267136, 19.833549 ],
+      zoom: 3
+    })
+
+    const tiles = L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      maxZoom: 18,
+      minZoom: 3,
+      attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
+    });
+
+    tiles.addTo(this.map);
+  }
+
+  ngAfterViewInit(): void {
+    this.initMap();
+  }
+
+  ngOnInit(): void {
+    this.refreshRoutes();
+  }
+  ngOnChanges(): void {
+    this.refreshRoutes();
+  }
+
+  route(): void {
+    if(this.map !== undefined &&
+       this.departureCoordinates !== undefined &&
+        this.destinationCoordinates !== undefined){
+      
+      this.routeControl = L.Routing.control({
+        waypoints: [this.departureCoordinates, this.destinationCoordinates],
+        lineOptions: {
+          styles: [{color: 'blue', opacity: 1, weight: 5}],
+          extendToWaypoints: false,
+          missingRouteTolerance: 0,
+        },
+        routeWhileDragging: false
+      }).addTo(this.map);
+      this.routeControl.show();     
+    }
+  }
+
+  private refreshActiveDrivers() {
+    this.mapService.getActiveDriversLocations().subscribe({
+      next: (locations) => {
+        this.driverLocationMarkers.forEach(driverMarker => {
+          this.map?.removeLayer(driverMarker);
```

Comment:
Why does each driver get his own layer? There should be one layer for all driver icons. Check the tutorial https://leafletjs.com/examples/layers-control/